lib/ts/recipe/thirdparty/components/themes/signInAndUp/providersForm.tsx

```diff
@@ -29,23 +29,39 @@ export const ThirdPartySignInAndUpProvidersForm: React.FC<SignInAndUpThemeProps>
 
     const signInClick = async (providerId: string): Promise<void> => {
         try {
-            const response = await redirectToThirdPartyLogin({
-                recipeImplementation: props.recipeImplementation,
-                thirdPartyId: providerId,
-                config: props.config,
-                userContext,
-            });
-            if (response.status === "ERROR") {
-                return props.dispatch({ type: "setError", error: "SOMETHING_WENT_WRONG_ERROR" });
+            let response;
+            let generalError: STGeneralError | undefined;
+
+            try {
+                response = await redirectToThirdPartyLogin({
+                    recipeImplementation: props.recipeImplementation,
+                    thirdPartyId: providerId,
+                    config: props.config,
+                    userContext,
+                });
+            } catch (e) {
+                if (STGeneralError.isThisError(e)) {
+                    generalError = e;
+                } else {
+                    throw e;
+                }
             }
-        } catch (err) {
-            if (STGeneralError.isThisError(err)) {
+
+            if (generalError !== undefined) {
                 return props.dispatch({
```

Comment:
Well.. you are still having to `return` from these types of general errors from the function which kind of beats the point of this refactor (whose original purpose was not to return if the older general error handling didn't return). If we must always return, then I prefer it being in the try catch like how it was done before this PR..

So overall, the point of this is that:
- If previously we were handling general error by checking the status, that code flow should be unchanged and only the way we detected if it was a general error would change.
- If previously we didn't have a general error status for an API call, we should now handle it in the try / catch block.

If you want to maintain that we handle general error in try catch everywhere, you need to make sure that if previously we handled general error without a return, the try / catch doesn't have a return then either.

Reply:
Looking at it again, this did not need to return early. Refactored it to handle it the same way it is done in other places